lib/screens/v2/disruption_diagram/model.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Screens.V2.DisruptionDiagram.Model do
+  @moduledoc """
+  Struct and functions to generate and model a disruption diagram.
+  """
+
+  # Model fields TBD
+  defstruct []
+
+  @type t :: %__MODULE__{}
+
+  @type serialized_response :: continuous_disruption_diagram() | discrete_disruption_diagram()
+
+  @type continuous_disruption_diagram :: %{
+          effect: :shuttle | :suspension,
+          # A 2-element list, giving indices of the effect region's *first and last disrupted stops*.
+          # For example in this scenario:
+          #     0     1     2     3     4     5     6     7     8
+          #    <= === O ========= O - - X - - X - - X - - O === =>
+          #                             |---range---|
+          # The range is [4, 6].
+          effect_region_slot_index_range: list(non_neg_integer()),
+          line: line_color(),
+          current_station_slot_index: non_neg_integer() | nil,
```

Comment:
A question I just thought about that I want to squeeze in. Why can this be `nil`?